shm/Ship.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <string>
+
+#include "Cargo.hpp"
+
+class Ship {
+public:
+    Ship()
+        : capacity_()
+        , maxCrew_()
+        , crew_()
+        , speed_()
+        , name_()
+        , cargo_()
+        , id_(-1)
+    {}
+
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
+        : capacity_(capacity)
+        , maxCrew_(maxCrew)
+        , crew_(0)
+        , speed_(speed)
+        , name_(name)
+        , cargo_()
+        , id_(id)
+    {}
+    
+    Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id)
+    {}
+
+    void setName(const std::string& name) { name_ = name; }
+
+    Ship& operator-=(size_t num) {
+        crew_ -= num;
+        return *this;
+    }
+    Ship& operator+=(size_t num) {
+        crew_ += num;
+        return *this;
+    }
```

Comment:
What happened when we reach the limit of capacity or drop below 0? You need to handle error cases. Maybe throw exception, maybe return `std::nullopt` or `nullptr` maybe set some specific value like `std::numeric_limits::min/max`.